### PR TITLE
feat(sync-tools): add method to wait until offer has exited

### PR DESCRIPTION
**Pull request by Jorge-Lopes**
closes: https://github.com/Agoric/BytePitchPartnerEng/issues/38

## Description

This PR aims to extend the available `sync-tools` by providing a method that allows the developer to verify that an offer was successfully exited via the execution of a `tryExitOffer` call.

The method `waitUntilOfferExited` will poll the user's wallet `liveOffers` and check if the respective `offerId` is no longer present.

### Security Considerations

No new assumptions or dependencies were introduced.

### Scaling Considerations

No requirements for increased resources.

### Documentation Considerations

This additional tool does not impact backward compatibility.

Although it may be useful to inform the developer that intends to use this method to prevent a false positive, it should first verify that the offerID is indeed present in the user's wallet liveOffers before calling `tryExitOffer`.

### Testing Considerations

I have confirmed that this method works as expected when used locally at the a3p-integration acceptance tests.
The test used is not included in the PR, since it would be out of scope.

### Upgrade Considerations

No impact is expected for live production systems.

### CODE REVIEW

File: packages/client-utils/src/sync-tools.js [hunk 1]
```diff
@@ -287,3 +288,46 @@ export const waitUntilInvitationReceived = (addr, io, options) => {
...
+  return !liveOffers.some(element => element.includes(offerId));
+};
+
+/**
+ *
+ * @param {string} addr
+ * @param {string} offerId
+ * @param {{ follow: () => object, log: typeof console.log, setTimeout: typeof global.setTimeout}} io
+ * @param {WaitUntilOptions} options
+ * @returns
```
Comment (Chris-Hibbert) [inline packages/client-utils/src/sync-tools.js]:
```suggestion
 * @param {string} addr
 * @param {string} offerId
 * @param {{ follow: () => object, log: typeof console.log, setTimeout: typeof global.setTimeout}} io
 * @param {WaitUntilOptions} options
```

File: packages/client-utils/src/sync-tools.js [hunk 2]
```diff
@@ -287,3 +288,46 @@ export const waitUntilInvitationReceived = (addr, io, options) => {
...
   );
 };
+
+/// ////////// Making sure an offer was exited successfully /////////////
+
+const makeQueryWalletCurrent = follow => async (/** @type {string} */ addr) => {
+  const update = await follow('-lF', `:published.wallet.${addr}.current`);
+
+  return update;
+};
```
Comment (Chris-Hibbert) [inline packages/client-utils/src/sync-tools.js]: I don't think there is any difference between the above and this shorter version.
```suggestion
const makeQueryWalletCurrent = follow => (/** @type {string} */ addr) =>
  follow('-lF', `:published.wallet.${addr}.current`);
```

File: packages/client-utils/src/sync-tools.js [hunk 3]
```diff
@@ -287,3 +288,46 @@ export const waitUntilInvitationReceived = (addr, io, options) => {
...
+
+/**
+ *
```
Comment (Chris-Hibbert) [inline packages/client-utils/src/sync-tools.js]: Please drop the blank line.

**Review by Chris-Hibbert — APPROVED** [review]

File: packages/client-utils/src/sync-tools.js (hunk 1 above)
Comment (Jorge-Lopes) [reply packages/client-utils/src/sync-tools.js]: Change request addressed

File: packages/client-utils/src/sync-tools.js (hunk 3 above)
Comment (Jorge-Lopes) [reply packages/client-utils/src/sync-tools.js]: Change request addressed

File: packages/client-utils/src/sync-tools.js (hunk 2 above)
Comment (Jorge-Lopes) [reply packages/client-utils/src/sync-tools.js]: Change request addressed

**Pull request merged**